app/controllers/ally_controller.rb

```diff
@@ -0,0 +1,142 @@
+class AllyController < ApplicationController
+
+  #
+  # REST methods
+  #
+
+  ##
+  # Returns whether the plugin is enabled for the account. If its
+  # enabled extra information such as the client ID and base URL
+  # of the Ally API is returned as well
+  def enabled
+    plugin_data = get_plugin_data
+    data = {
+      :enabled => plugin_data[:enabled],
+      :clientId => plugin_data[:client_id],
+      :baseUrl => plugin_data[:base_url]
+    }
+    render :json => data
+  end
+
+  ##
+  # Sign a request to the Ally REST API but don't execute it.
+  # The UI will execute the request
+  def sign
+    if (request = get_request())
+      plugin_data = get_plugin_data()
+      puts request.to_yaml
+      data = {
+        :clientId => plugin_data[:client_id],
+        :baseUrl => plugin_data[:base_url],
+        :path => request.path,
+        :header => request.to_hash["authorization"][0],
+        :body => request.body,
+      }
+      render :json => data
+    end
+  end
+
+  ##
+  # Proxy data from the Ally REST API
+  def proxy
+    if (request = get_request())
+      # TODO: Cache the HTTP object per hostname allowing
+      # for connection pooling
```

Comment:
I ask because normally standard libs don't provide pooling support, and it's usually a lot more complicated than caching a connection. I'd like to avoid implying an implementation at this point.

There is something like this that could be investigated: https://github.com/mperham/connection_pool . Or maybe canvas already has some http connection pooling client support available.